src/gfloat/types.py

```diff
@@ -201,7 +206,12 @@ def min(self) -> float:
         """
         The smallest representable number, typically ``-max``.
         """
-        return -self.max
+        if self.is_signed:
+            return -self.max
```

Comment:
A note to your future self that this won't be true if you later add ocp_int8 as it can be asymmetric.

Reply:
There is a test that `code_of_min` decodes to `min`, and `code_of_min` is "self-evidently" 2**k-1, so it should be fine, but...  I have now added a dense test to this PR which exhaustively enumerates all values and ensures min, max, etc are consistent.

Reply:
I forgot to mention: this PR does include OCP int8, but I let it decode 0x80 as a NaN, which is allowed by [at least one reading of] the spec...
> "Integer data types use a 2’s complement encoding, but the maximum negative representation
(−2) may be left unused to maintain symmetry between the maximum positive and negative
representations and avoid introducing a negative bias."

But maybe that is to be read as "0x80 should decode to -2, and downstream users can choose not to use it...."

Reply:
I'll do it that way, as the spec says "N/A" for NaN, but presumably means "not available" rather than "not applicable".